File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/LMS/ClassroomSubject.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.LMS
{
    public class ClassroomSubject : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        public bool Hide { get; set; }

        [ForeignKey("Teacher")]
        public long? TeacherID { get; set; }
        public Employee? Teacher { get; set; }
        
        [ForeignKey("Classroom")]
        public long ClassroomID { get; set; }
        public Classroom Classroom { get; set; }

        [ForeignKey("Subject")]
        public long SubjectID { get; set; }
        public Subject Subject { get; set; }

        public ICollection<ClassroomSubjectCoTeacher> ClassroomSubjectCoTeachers { get; set; } = new HashSet<ClassroomSubjectCoTeacher>();
    }
}

File: LMS_CMS_Frontend/src/app/Models/LMS/classroom-subject.ts
import { ClassroomSubjectCoTeacher } from "./classroom-subject-co-teacher";

export class ClassroomSubject {
    constructor(
        public id: number = 0,
        public hide: boolean = false,
        public teacherID: number = 0,
        public teacherEnglishName: string = '',
        public teacherArabicName: string = '',
        public subjectID: number = 0,
        public subjectEnglishName: string = '',
        public subjectArabicName: string = '', 
        public classroomID: number = 0,
        public insertedByUserId: number = 0,
        public coTeacherIDs :number[] = [],
        public classroomSubjectCoTeachers :ClassroomSubjectCoTeacher[] = [],
    ) {}
} 